pkg/sql/codegen/optimize/template_optimize.go

```diff
@@ -75,27 +83,36 @@ def submit():
                          partitions=None)
     
     runner = RunnerConf(cls="{{.RunnerModule}}.CustomOptFlowRunner")
-    
+
+    data_options = attributes.get("data", {})
+    sys.stderr.write('data options: {}\n'.format(data_options))
+    enable_slice = data_options.get("enable_slice", False)
+    batch_size = data_options.get("batch_size", None)
+    if batch_size <= 0:
+         batch_size = None
+
     output_table = OdpsItemConf(path="odps://{{.ResultTable}}", odps=odps_conf)
     output = OutputConf(df1=output_table)
 
     df1 = OdpsItemConf(path="odps://{{.TrainTable}}",
                        odps=odps_conf,
-                       enable_slice=False)
+                       enable_slice=enable_slice,
+                       batch_size=batch_size)
     
     input_conf = InputConf(df1=df1)
 
     optflow_version = os.environ.get("SQLFLOW_OPTFLOW_VERSION")
     if not optflow_version:
         raise ValueError("Environment variable SQLFLOW_OPTFLOW_VERSION must be set")
     	
-    if "{{.IsPAI}}":
+    if is_pai:
```

Comment:
check worker count > 1. `is_pai` should not be used in optimize

Reply:
As offline discussion, `is_pai` is removed.